### PR TITLE
Feat/standalone build

### PR DESCRIPTION
**Pull request by logan-anderson**
Add a "static build to dev and build scripts the the correct config is provided.

## How to setup
 Here is how to setup on a new project. This can be in the monorepo (the basic starter has not been setup yet) or can be an external with the a tag (`yarn add tinacms@standalone-build` and `yarn add --dev @tinamcs/cli@standalone-build`).
 1. Add a `.tina/config.{js,ts,tsx,jsx}` with the default export of define config.
```ts
// .tina/config.ts
import schema from './schema'

export default defineConfig({
  schema: schema,
  //.. Everything from define config in `schema.ts`
  //.. Everything from `schema.config`  
}) 
```

2. Add Build config

```
.tina/config.ts

export default defineConfig({
   build: {
     outputFolder: "tina",
     publicFolder: "public",
  },
  //... other config
})
```

3. Update `.tina/components/TinaProvider.jsx`

```diff
import TinaCMS from "tinacms";
+ import config from "../config";
- import { config }  from "../schema";

// Importing the TinaProvider directly into your page will cause Tina to be added to the production bundle.
// Instead, import the tina/provider/index default export to have it dynamially imported in edit-moode
/**
 *
 * @private Do not import this directly, please import the dynamic provider instead
 */
const TinaProvider = ({ children }) => {
  return <TinaCMS {...config}>{children}</TinaCMS>;
};

export default TinaProvider;
```

4. Go to `http://localhost:300/tina/index.html` and view the admin

## What to test
-  Go to `http://localhost:300/tina/index.html` 
- Try to edit with the "View in admin view"
- Try to view and edit different types of forms (Text, Images, date, etc)

### CODE REVIEW

**Comment by jeffsee55** [comment]
> 3. Update .tina/components/TinaProvider.jsx


What's the purposed of step 3? As far as I know step 3 should not be necessary at all, and we can actually go ahead and remove the `components` directory entirely.

**Comment by jamespohalloran** [comment]
> > 3. Update .tina/components/TinaProvider.jsx
> 
> What's the purposed of step 3? As far as I know step 3 should not be necessary at all, and we can actually go ahead and remove the `components` directory entirely when we land this for all uses.

I believe step 3 is so that You can use this standalone approach with contextual editing (Seems like this may go away with the iframe approach later though). If this is the case, should part of the setup also include:
- Remove pages/admin.js
- Setup a nextjs rewrite ("/tina/index.html" -> "/admin") so that links from contextual-editing to the admin continue to work?

**Comment by logan-anderson** [comment]
> I believe step 3 is so that You can use this standalone approach with contextual editing (Seems like this may go away with the iframe approach later though). If this is the case, should part of the setup also include:

Yup exactly @jamespohalloran

**Review by jamespohalloran — COMMENTED** [review]
Left a few comments, feels good overall though! 

A few questions:
- I see we're calling the admin page "tina" instead of "admin", is that to keep things backwards compatible incase users already have an "admin" page with Tina?
- The vite static build step does feel a tad slow, are there potential caching upgrades that could help?

**Comment by jeffsee55** [comment]
> You can use this standalone approach with contextual editing

I guess if we're not documenting this it's harmless but seems confusing to use both.

> The vite static build step does feel a tad slow

We are able to use the Vite dev server which is super fast but it requires that we mess with your rewrites. Definitely something we can do if this is too slow but maybe we can leave it for now?

**Comment by jamespohalloran** [comment]
> it requires that we mess with your rewrites. 

What do you mean by that?

>...Definitely something we can do if this is too slow but maybe we can leave it for now?

If this was an opt-in feature, I'd say it can wait. It seems like it's going to impact the performance for all sites though which might be a problem

File: packages/@tinacms/toolkit/package.json [hunk 1]
```diff
@@ -112,6 +112,7 @@
     "react-dropzone": "10.1.8",
     "react-final-form": "^6.3.0",
     "react-icons": "^4.3.1",
+    "react-onclickoutside": "^6.12.2",
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply packages/@tinacms/toolkit/package.json:115]: Yes it is.

File: packages/@tinacms/app/appFiles/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply packages/@tinacms/app/appFiles/index.html:7]: Definitely I will update the PR

File: .changeset/tasty-walls-hang.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+2. Add Build config
+
+```
+.tina/config.ts
+
+export default defineConfig({
+   build: {
+     outputFolder: "tina",
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply .changeset/tasty-walls-hang.md:37]: Sounds good! I will update the changeset .

**Comment by logan-anderson** [comment]
@jamespohalloran 
> If this was an opt-in feature, I'd say it can wait. It seems like it's going to impact the performance for all sites though which might be a problem:
> Edit: Maybe I'm wrong on that. If so, can we clarify what the opt-in steps are?


This is an opt in feature. It will be opt in if you have the outputFolder and publicFolder set in `defineConfig` in `.tina/config.{ts,js}`

> What do you mean by that?

The user will have to setup [Next.js redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) to send all requests from '/admin' to the vite dev server (not part of the PR).

**Comment by jamespohalloran** [comment]
> @jamespohalloran
> 
> > If this was an opt-in feature, I'd say it can wait. It seems like it's going to impact the performance for all sites though which might be a problem:
> > Edit: Maybe I'm wrong on that. If so, can we clarify what the opt-in steps are?
> 
> This is an opt in feature. It will be opt in if you have the outputFolder and publicFolder set in `defineConfig` in `.tina/config.{ts,js}`
> 
> > What do you mean by that?
> 
> The user will have to setup [Next.js redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) to send all requests from '/admin' to the vite dev server (not part of the PR).

Gotcha, yeah in that case I think the build speed improvements don't need to hold this PR up.

**Comment by logan-anderson** [comment]
PR and npm tag have been update based on your comments @jamespohalloran

**Review by jamespohalloran — APPROVED** [review]
Cool, lgtm!

**Pull request merged**